lib/src/analysis_server.dart

```diff
@@ -234,31 +234,45 @@ abstract class AnalysisServerWrapper {
       await _loadSources(<String, String>{mainPath: source});
       await analysisCompleter.future;
 
-      final result = await analysisServer.analysis.getHover(mainPath, offset);
+      final result = await analysisServer!.analysis.getHover(mainPath, offset);
       await _unloadSources();
 
       if (result.hovers.isEmpty) {
-        return null;
+        return <String, String>{};
```

Comment:
Is this map added to later on? Perhaps this could be made const?

```suggestion
        return const {};
```